packages/@aws-cdk/aws-appconfig-alpha/lib/environment.ts

```diff
@@ -276,21 +284,57 @@ export class Environment extends EnvironmentBase {
   }
 }
 
+export enum MonitorType {
+  CLOUDWATCH,
+  CFN_MONITORS_PROPERTY,
+}
+
 /**
  * Defines monitors that will be associated with an AWS AppConfig environment.
  */
-export interface Monitor {
+export abstract class Monitor {
   /**
-   * The Amazon CloudWatch alarm.
-   */
-  readonly alarm: cloudwatch.IAlarm;
+   * Creates a Monitor from a CloudWatch alarm. If the alarm role is not specified, a role will
+   * be generated.
+   *
+   * @param alarm The Amazon CloudWatch alarm.
+   * @param alarmRole The IAM role for AWS AppConfig to view the alarm state.
+   */
+  public static fromCloudWatchAlarm(alarm: cloudwatch.IAlarm, alarmRole?: iam.IRole): Monitor {
+    return {
+      alarmArn: alarm.alarmArn,
+      alarmRoleArn: alarmRole?.roleArn,
+      monitorType: MonitorType.CLOUDWATCH,
+    };
+  }
 
   /**
-   * The IAM role for AWS AppConfig to view the alarm state.
+   * Creates a Monitor from a CfnEnvironment.MonitorsProperty construct.
    *
-   * @default - A role is generated.
+   * @param monitorsProperty The monitors property.
+   */
+  public static fromCfnMonitorsProperty(monitorsProperty: CfnEnvironment.MonitorsProperty): Monitor {
+    return {
+      alarmArn: monitorsProperty.alarmArn!,
+      alarmRoleArn: monitorsProperty.alarmRoleArn,
+      monitorType: MonitorType.CFN_MONITORS_PROPERTY,
+    };
+  }
+
+  /**
+   * The alarm ARN for AWS AppConfig to monitor.
+   */
+  public abstract readonly alarmArn: string;
+
+  /**
+   * The IAM role ARN for AWS AppConfig to view the alarm state.
+   */
+  public abstract readonly alarmRoleArn?: string;
```

Comment:
nit: this optional class variable is defined before a required one.

Reply:
Updated